File: src/main/java/ru/citeck/idea/references/resolvers/JsonStringLiteralValueResolver.java
package ru.citeck.idea.references.resolvers;

import com.intellij.json.psi.JsonStringLiteral;
import ru.citeck.idea.references.PsiElementValueResolver;

public class JsonStringLiteralValueResolver implements PsiElementValueResolver<JsonStringLiteral> {

    @Override
    public Class<JsonStringLiteral> getPsiElementType() {
        return JsonStringLiteral.class;
    }

    @Override
    public Object getValue(JsonStringLiteral psiElement) {
        return psiElement.getValue();
    }
}

File: src/main/kotlin/ru/citeck/idea/references/resolvers/JsonStringLiteralValueResolver.kt
package ru.citeck.idea.references.resolvers

import com.intellij.json.psi.JsonStringLiteral
import ru.citeck.idea.references.PsiElementValueResolver

class JsonStringLiteralValueResolver : PsiElementValueResolver<JsonStringLiteral> {

    override fun getPsiElementType(): Class<JsonStringLiteral> {
        return JsonStringLiteral::class.java
    }

    override fun getValue(psiElement: JsonStringLiteral): Any {
        return psiElement.value
    }
}
